src/test/java/strategy/impl/PurchaseTypeHandlerTest.java

```diff
@@ -0,0 +1,28 @@
+package strategy.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.Test;
+
+public class PurchaseTypeHandlerTest {
+
+    private static final PurchaseTypeHandler PURCHASE_TYPE_HANDLER = new PurchaseTypeHandler();
```

Comment:
In Java, constant naming conventions (using all uppercase letters with underscores) are typically reserved for immutable values, such as static final primitive values or immutable objects.

The PURCHASE_TYPE_HANDLER instance is not an immutable constant in the truest sense, even though it is marked as final. It refers to an instance of a class, and unless the class itself is designed to be immutable, the object could potentially have mutable state.

A better convention for this case would be to use camelCase for the field name.